mllib/src/main/scala/org/apache/spark/ml/classification/RandomForestClassifier.scala

```diff
@@ -134,13 +136,28 @@ final class RandomForestClassificationModel private[ml] (
 
   override def treeWeights: Array[Double] = _treeWeights
 
+  override def transform(dataset: DataFrame): DataFrame = {
+    val bcastModel = dataset.sqlContext.sparkContext.broadcast(this)
+    transformImpl(dataset, bcastModel)
+  }
+
   override protected def predict(features: Vector): Double = {
-    // TODO: Override transform() to broadcast model.  SPARK-7127
     // TODO: When we add a generic Bagging class, handle transform there: SPARK-7128
+    // Predict without using a broadcasted mode
+    predictImpl(features, () => this)
```

Comment:
Yes, we will want to.  If it's a small model, then it can be faster to not broadcast it.